hack/make.ps1

```diff
@@ -256,20 +256,12 @@ Function Validate-PkgImports($headCommit, $upstreamCommit) {
     $files=@(); $files = Invoke-Expression "git diff $upstreamCommit...$headCommit --diff-filter=ACMR --name-only -- `'pkg\*.go`'"
     $badFiles=@(); $files | ForEach-Object{
         $file=$_
-        if ($file -eq "pkg\urlutil\deprecated.go") {
-            # pkg/urlutil is deprecated, but has a temporary alias to help migration,
-            # see https://github.com/moby/moby/pull/43477
-            # TODO(thaJeztah) remove this exception once pkg/urlutil aliases are removed
-            return
-        }
         # For the current changed file, get its list of dependencies, sorted and uniqued.
         $imports = Invoke-Expression "go list -e -f `'{{ .Deps }}`' $file"
         if ($LASTEXITCODE -ne 0) { Throw "Failed go list for dependencies on $file" }
         $imports = $imports -Replace "\[" -Replace "\]", "" -Split(" ") | Sort-Object | Get-Unique
         # Filter out what we are looking for
-        $imports = @() + $imports -NotMatch "^github.com/docker/docker/pkg/" `
-                                  -NotMatch "^github.com/docker/docker/vendor" `
-                                  -Match "^github.com/docker/docker" `
+        $imports = @() + $imports -Match "^github.com/docker/docker/(cli|cmd)" `
```

Comment:
I wonder if instead of our own crafted check, if we could replace it with `depguard` 🤔 (with some proper configuration of that) https://github.com/moby/moby/blob/058010187bdae5364d82c4cff204f103231b13e0/hack/validate/golangci-lint.yml#L29-L35

(we can probably remove that specific check though, as `io/ioutils` is marked deprecated, so should probably already show up in other linters)

Reply:
https://golangci-lint.run/usage/linters#depguard
Yeah, it does look feasible with something like
```yaml
linters-settings:
  depguard:
    additional-guards:
      - list-type: denylist
        packages:
          - github.com/docker/docker/cmd/**
          - github.com/docker/docker/cli/**
          - github.com/docker/docker/daemon/**
        ignore-file-rules:
          - "!pkg/**/*.go"
```

Reply:
Ah, nice! Yes, if it works, I like that approach; it's more visible and easier to understand to others (without having to dig into obscure shell scripts)